File: fx_ext/background/logger.js
class logger {
    static logs = [];
    static log(...args) {
        var l = `[${new Date().toLocaleTimeString('zh',{hour12:false})}] ${args.join(' ')}`;
        if (this.logger) {
            this.logger.info(l);
        } else {
            this.logs.push(l + '\n');
            console.log(l);
        }
    }
    static info(...args) {
        this.log(...args);
    }
    static error(...args) {
        this.log(...args);
    }
    static debug(...args) {
        this.log(...args);
    }
}

File: iun/app/logger.py
import os, sys
import logging


class LazyLogger:
    def __init__(self):
        self._logger = None
        
    def _init_logger(self):
        if self._logger is None:
            lg_path = os.path.join(os.path.dirname(__file__), '../logs/iun.log')
            logging.basicConfig(
                level=logging.INFO,
                format='[%(levelname)s] %(asctime)s-%(name)s: %(message)s',
                handlers=[logging.FileHandler(lg_path), logging.StreamHandler(sys.stdout)]
            )
            self._logger = logging.getLogger('iun')
    
    def __getattr__(self, name):
        self._init_logger()
        return getattr(self._logger, name)

logger = LazyLogger()
